api/src/neo4j/queries/reaction.js

```diff
@@ -47,13 +47,15 @@ CALL apoc.cypher.run("
   
   MATCH (r:Reaction${m} {id: '${id}'})-[${v}]-(cm:CompartmentalizedMetabolite)
   WITH DISTINCT cm
-  MATCH (cm)-[${v}]-(:Compartment)-[${v}]-(csvg:SvgMap)
-  RETURN { compartmentSVGs: COLLECT(DISTINCT(csvg {.*})) } as data
+  MATCH (cm)-[${v}]-(c:Compartment)-[${v}]-(csvg:SvgMap)
+  WITH {compartmentId: c.id, compartmentSVGs: COLLECT(DISTINCT(csvg {.*}))} as compartmentSVG
+  RETURN { compartmentSVGs: COLLECT(compartmentSVG) } as data
```

Comment:
line 17 in `/frontend/src/store/modules/reactions.js`. Basically a fix to be able to group multiple maps belonging to same compartment, such as cytosol

Reply:
I think the `compartmentId: c.id` can be skipped to end up with
```suggestion
  WITH {compartmentSVGs: COLLECT(DISTINCT(csvg {.*}))} as compartmentSVG
  RETURN { compartmentSVGs: COLLECT(compartmentSVG) } as data
```

Reply:
@mihai-sysbio that is almost what we had before. I think the change @inghylt made is great here because there is not enough info in the `compartmentSVG` nodes to reliably associate an SVG map with a compartment or subsystem. While it is currently technically possible to parse the compartment name out of something like `cytosol_1`, I think @inghylt's implementation is much more reliable since it's directly based on the graph database structure.

Reply:
Now I see, thanks for the demo @inghylt . `compartmentSVG ` is not the same as `compartmentSVGs`, my bad!